src/System.Management.Automation/engine/remoting/server/ServerRunspacePoolDriver.cs

```diff
@@ -1392,126 +1423,102 @@ private class DebuggerCommandArgument
                 // Input parameters:
                 // [-Id <int>]
                 // Returns Breakpoint object(s).
-                string script = null;
-
-                if (command.Parameters?.Count > 0)
+                TryGetParameter<int?>(command, "RunspaceId", out int? runspaceId);
+                if (TryGetParameter<int>(command, "Id", out int breakpointId))
                 {
-                    int breakpointId = CheckBreakpointIdParameter(command);
-                    script = $"$host.Runspace.Debugger.GetBreakpoint({breakpointId})";
+                     preProcessOutput.Add(serverRemoteDebugger.GetBreakpoint(breakpointId, runspaceId));
                 }
                 else
                 {
-                    script = $"$host.Runspace.Debugger.GetBreakpoints()";
+                    foreach (Breakpoint breakpoint in serverRemoteDebugger.GetBreakpoints(runspaceId))
+                    {
+                        preProcessOutput.Add(breakpoint);
+                    }
                 }
 
-                ReplaceVirtualCommandWithScript(commands, script);
+                result = PreProcessCommandResult.BreakpointManagement;
             }
             else if (commandText.Equals(RemoteDebuggingCommands.SetBreakpoint, StringComparison.OrdinalIgnoreCase))
             {
                 // __Set-PSBreakpoint private virtual command.
                 // Input parameters:
-                // [-Script] <string[]> [-Line] <int[]> [[-Column] <int>] [-Action <scriptblock>]
-                // [[-Script] <string[]>] -Command <string[]> [-Action <scriptblock>]
-                // [[-Script] <string[]>] -Variable <string[]> [-Action <scriptblock>] [-Mode <VariableAccessMode>]
-                // Returns Breakpoint object.
-                if (command.Parameters == null || command.Parameters.Count == 0)
+                // -Breakpoint <Breakpoint> or -BreakpointList <IEnumerable<Breakpoint>>
+                // [-RunspaceId <int?>]
+                // Returns Breakpoint object(s).
+                TryGetParameter<Breakpoint>(command, "Breakpoint", out Breakpoint breakpoint);
```

Comment:
The `SetBreakpoints()` API passes a `Breakpoint` array type, so this will always return null and remoting SetBreakpoints will always fail.  I think we need to get a parameter of `Breakpoint[]` type.

Reply:
yep. A typo in a parameter. Will fix.